bleak/backends/scanner.py

```diff
@@ -141,6 +141,9 @@ def __init__(
         )
 
         self.seen_devices = {}
+        self.new_devices: Optional[
```

Comment:
I would much rather not add new state here as managing state like this has been the main cause of bugs in Bleak.

Reply:
This was done as simplest/most-effective workaround to avoid https://github.com/hbldh/bleak/pull/1361/files#diff-e4f68248febf58b848a3fd5bbdce9a669c236349f58b7d9005b704fd507ab8b6R145-R149 plus https://github.com/hbldh/bleak/pull/1361/files#diff-a1877ebc56efdeaa2019b7e6738a9eba73a8788f646396e9a6221ac36f2d9f83R218-R221.
I had 2 implementations before this: the one implemented now and "the most proper one" from the user standpoint, which saved user callback passed to `BleakScanner(detection_callback)` to internal variable, registered own `BleakScanner._detection_calback()` in `self._backend = PlatformBleakScanner(self._detection_calback, ...)` and then called the user callback, if provided, from this internal callback after adding the device to the queue. However, this meant that I fully replicated the `BaseBleakScanner`'s `detection_callback` logic (https://github.com/hbldh/bleak/pull/1361/files#diff-e4f68248febf58b848a3fd5bbdce9a669c236349f58b7d9005b704fd507ab8b6R150-R184), which I did not like.

The solution now bases on the fact that: if a user is using the scanner as async iterator, it shall not require the `detection_callback` functionality, as coroutines are meant to replace this ("callback hell") exactly.